### ISSUE TITLE
DataFrame distinct with multiple columns in the selector is not working if the name of the columns is variable

### ISSUE THREAD

**Issue by marcodisa** [data-forge/data-forge-ts#155]
Hi, I'm not sure if this is the proper section to post this question,
First of all I’d like to congratulate you on the amazing work you’ve done with Data Forge npm and Data Forge Notebook. It’s a very extraordinary package and the Notebook is an extremely useful tool.
I’m seriously considering the upgrade to the pro version for my further projects.

Anyway, I’m stuck on a DataFrame distinct with two parameters in the selector.

If I run the code below it perfectly does what I want, “filiale” and “CAP” are two columns of df.

```
const safeJoinChar = '$';
var distinct_join = df.distinct(value => [value.filiale, value.CAP].join(safeJoinChar));
console.log(distinct_join.toString());
```

Actually I need to put the execution of the select distinct into a loop. I need the select distinct of every column with all the other columns, two columns at a time. 
Long story short, I have a loop that in every iteration assigns the value to two variables serie1_name and serie2_name, in every iteration I correctly get my two variables like the code below. 
Now I need the dataframe distinct to work with variable names not with the explicit names of the columns. 
How the code below should be to work fine with variable column names?

```
var serie1_name = "filiale";
var serie2_name = "CAP";
const safeJoinChar = '$';

var distinct_join = df.distinct(value => [value.serie1_name, value.serie2_name].join(safeJoinChar));

console.log(distinct_join.toString());
```

I’ve tried many ways to make it work but I wasn’t able to handle it myself.
Is there something I’m missing out?
Can I get any hints or suggestions?

Thank you very much!

**Comment by marcodisa** [comment]
After many attempts I solved the issue using the javascript function conscrtuctor to create a function from a string

```
var serie1_name = "filiale";
var serie2_name = "CAP";
const safeJoinChar = '$';

var distinctFormula = new Function(serie1_name, serie2_name, safeJoinChar, "return value => [value."+ serie1_name + ", value." + serie2_name + "].join("+safeJoinChar+")");

var distinct_join = data.distinct(distinctFormula());

console.log(distinct_join.toString());
```

Now it works perfectly!

Anyway, is there a more clever solution?

**Comment by ashleydavis** [comment]
That's strange. You shouldn't have to use a string.

Are you able to publish a runnable code example in GitHub of the broken version? Then I can take a look at it.

**Comment by ashleydavis** [comment]
Hey @marcodisa any chance you could fork the repo then get your code working as a failing test? Then I can work towards fixing the code for you and making your test pass.
